client/src/rpc_client.rs

```diff
@@ -128,12 +128,32 @@ impl RpcClient {
         self.send_transaction_with_config(transaction, RpcSendTransactionConfig::default())
     }
 
+    fn cluster_transaction_encoding(&self) -> Result<UiTransactionEncoding, RpcError> {
```

Comment:
Ack, so we're going to need to cache the version or the rate limiter is going to get really mad during `solana deploy` as the RPC calls 2x.

1. `cluster_transaction_encoding()` -> `default_cluster_transaction_encoding()`?
2. Cache in `self` on the first call, and reuse it for the lifetime of the `RpcClient`.   This is slightly disingenuous since the endpoint could change versions on us but that's pretty unlikely.

Reply:
Renamed and cached!